src/main/java/com/puppycrawl/tools/checkstyle/api/AbstractCheck.java

```diff
@@ -39,15 +39,15 @@
     /** Default tab width for column reporting. */
     private static final int DEFAULT_TAB_WIDTH = 8;
 
+    /**
+     * The check context.
+     * @noinspection ThreadLocalNotStaticFinal
```

Comment:
@romani here (https://gist.github.com/sabaka/3fa88c9b601c3c8819244834abf1a07f) is a gist from @sabaka, if I understood you correctly. It shows the difference between static thread local and thread local member.